### ISSUE TITLE
"ValueError: Format specifier missing precision" on pipx install <package>

### ISSUE THREAD

**Issue by halloleo** [pipxproject/pipx#444]
**Describe the bug**

When running `pipx install cowsay` I get the following error:

```
pipx install cowsay
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipx/animate.py", line 78, in print_animation
    cur_line = f"{s} {message:.{max_message_len}}"
ValueError: Format specifier missing precision

Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipx/animate.py", line 78, in print_animation
    cur_line = f"{s} {message:.{max_message_len}}"
ValueError: Format specifier missing precision

done! ✨ 🌟 ✨
  installed package cowsay 2.0.3, Python 3.7.7
  These apps are now globally available
    - cowsay
```

So `cowsay` installs correctly, but something goes wrong before that.

**How to reproduce**

My environment is:

&nbsp; | &nbsp;
-- | -------
OS | macOS Mojave
Python | Version 3.7.7<BR>installed via `brew`<BR>(and pinned)
pipx | Version 0.15.4<BR>installed via `pip`<BR>straight into the main  Python 3.7 environment

**Comment by itsayellow** [comment]
Very interesting.  The only way I can think of this happening is if somehow `shutil.get_terminal_size()` is returning a very small number for the number of columns, which would make `max_message_len` become a negative number.

Can you try to open your python3.7 interactive prompt and execute the following, reporting back what it says, please?
```
import shutil
shutil.get_terminal_size()
```

**Comment by halloleo** [comment]
@itsayellow Thanks for chiming in! 

I have closed the shell I was in at the time. When I call `get_terminal_size()` now I get

> os.terminal_size(columns=101, lines=44)

And when I try `pipx install cowsay` now , it works without any error.  So this must have been something funny in my shell...

**Comment by cs01** [comment]
> Very interesting. The only way I can think of this happening is if somehow shutil.get_terminal_size() is returning a very small number for the number of columns, which would make max_message_len become a negative number.

We should code defensively for this to avoid this (rare) error

**Comment by halloleo** [comment]
Hey, I got the error again: This time I can check `shutil.get_terminal_size()` immediately and it reports

> os.terminal_size(columns=0, lines=0)

(This might happen, because my shell is the xonsh and/or because it runs inside an Emacs buffer.)

@cs01 I do agree that a normal CLI tool should not fail just when it cannot get the terminal size.

**Comment by itsayellow** [comment]
Yikes!  Thanks for the debug help.  That's great information

We do supply defaults that `shutile.get_terminal_size()` is supposed to use if it cannot determine terminal size, but I guess somehow it really think the terminal size here is (0,0).

Yes indeed this needs to be fixed.

**Comment by itsayellow** [comment]
WFM: inside of xonsh `shutil.get_terminal_size()` returns a proper non-zero terminal_size.

I'm not an emacs user, @halloleo how exactly do you run this inside an emacs buffer?  Is it like having an interactive shell inside of emacs?

**Comment by itsayellow** [comment]
OK, I have emacs fired up and an interactive shell.  I can't make it break get_terminal_size(), even inside of both emacs and xonsh.  I was trying to replicate it in the hopes of maybe submitting a bug report to shutil.

But it's still broken for you and that's enough to know pipx should fix it.

**Comment by halloleo** [comment]
I run xonsh in emacs on all of my machines (some macOS, some RHEL Linux), only on *one* machine I have *sometimes* this trouble. So it must be more than just the xonsh in emacs setup. Maybe some commands I ran beforehand? Hard to pin down.

**Comment by itsayellow** [comment]
Fixed with #453
